fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java

```diff
@@ -280,8 +280,15 @@ public static List<Constraint> getConstraints(String url, Class<?> type) {
 
     private static List<Constraint> getConstraints(StructureDefinition profile, Class<?> type) {
         String url = profile.getUrl().getValue();
-        String version = profile.getVersion().getValue();
-        CacheKey key = key(url + "|" + version);
+        CacheKey key;
```

Comment:
My suggestion. Use final to ensure key is assigned once and once only. Also some updates to match the style for this repo:
```
        final CacheKey key;
        if (profile.getVersion() != null) {
            key = key(url + "|" + profile.getVersion().getValue());
        } else {
            key = key(url);
        }

```

Reply:
I turned this into a github suggestion just below.  With this change, I'm happy to approve.  I'll apply it later if we don't here back.  Thanks @berkant-k !